src/pal/tools/gen-buildsys-gcc.sh

```diff
@@ -80,55 +82,33 @@ for i in "$@"; do
     esac
 done
 
-OS=`uname`
+OS=$(uname)
 
 locate_gcc_exec() {
   if command -v "$gcc_prefix$1$desired_gcc_version" > /dev/null 2>&1
   then
-    echo "$(command -v $gcc_prefix$1$desired_gcc_version)"
+    command -v "$gcc_prefix$1$desired_gcc_version"
   elif command -v "$gcc_prefix$1" > /dev/null 2>&1
   then
-    echo "$(command -v $gcc_prefix$1)"
+    command -v "$gcc_prefix$1"
   else
     exit 1
   fi
 }
 
-gcc_ar="$(locate_gcc_exec ar)"
-[ $? -eq 0 ] || { echo "Unable to locate gcc-ar"; exit 1; }
+if ! gcc_ar="$(locate_gcc_exec ar)"; then { echo "Unable to locate gcc-ar"; exit 1; } fi
 
-if [ -z "$CC" ]; then
-  gcc_link="$(locate_gcc_exec gcc)"
-  [ $? -eq 0 ] || { echo "Unable to locate gcc-link"; exit 1; }
-else
-  gcc_link="$CC"
-fi
+if ! gcc_link="$(locate_gcc_exec link)"; then { echo "Unable to locate gcc-link"; exit 1; } fi
```

Comment:
This needs to be:

if ! gcc_link="$(locate_gcc_exec gcc)"; then { echo "Unable to locate gcc"; exit 1; } fi

Reply:
Can you explain why is that? Neither it aligns with clang buildsys, nor it makes sense. I do not have any problem with it on Ubuntu and FreeBSD.